sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -25,7 +25,22 @@
 class BaseLoss(ABC):
     """Base class for a loss."""
 
-    def init_gradients_and_hessians(self, n_samples, prediction_dim):
+    @abstractmethod
+    def __call__(self, y_true, raw_predictions):
```

Comment:
I think all the `__call__` methods should take sample_weight into account (and we could get rid of `get_average_loss`).